src/ILCompiler.Compiler/src/Compiler/GeneratedOnlyMetadataGeneration.cs

```diff
@@ -0,0 +1,164 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+
+using Internal.IL.Stubs;
+using Internal.TypeSystem;
+using Internal.Metadata.NativeFormat.Writer;
+
+using ILCompiler.Metadata;
+using ILCompiler.DependencyAnalysis;
+
+namespace ILCompiler
+{
+    public class GeneratedOnlyMetadataGeneration : MetadataGeneration
```

Comment:
No, it was a not entirely horrible name I came up with. I'm happy with your name.